modules/cpmso-yaml-failure-domain-vsphere.adoc

```diff
@@ -0,0 +1,30 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cpmso-configuration.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="cpmso-yaml-failure-domain-vsphere_{context}"]
+= Sample VMware vSphere failure domain configuration
+
+On VMware vSphere infrastructure, a failure domain gets defined in an infrastructure resource specification. A failure domain is an infrastructure resource that comprises a control plane machine set, a vCenter datacenter, vCenter datastore, and a network. By using a failure domain resource, you can deploy control plane machines on hardware that is separate from the primary VMware vSphere infrastructure.
```

Comment:
Query to SME: The infrastructure resource specification is equivalent to the "provider specification"? See   [Sample install-config.yaml file for an installer-provisioned VMware vSphere cluster](https://docs.openshift.com/container-platform/4.14/machine_management/control_plane_machine_management/cpmso-configuration.html#cpmso-yaml-provider-spec-vsphere_cpmso-configuration)

Reply:
not quite, the infrastructure resource defines the failure domains, but the cpms spec specifies the names of the failure domains to use.

Reply:
Thanks for the reply. Is the `infrastructure resource specification` the same as the `install-config.yaml` [file](https://docs.openshift.com/container-platform/4.14/installing/installing_vsphere/installing-vsphere-installer-provisioned-network-customizations.html#installation-installer-provisioned-vsphere-config-yaml_installing-vsphere-installer-provisioned-network-customizations)?

Reply:
Both the `infrastructure resource specification` and CPMS `providerspec` use name parameters, so the CPMS file is where failure domains gets assigned to a region? Why does the `infrastructure resource specification` spec need to define a name for the failure domain?